src/simpleapi.jl

```diff
@@ -120,9 +129,19 @@ end
 
 function delete!(ctx::KuberContext, O::Symbol, name::String; kwargs...)
     isempty(ctx.apis) && set_api_versions!(ctx)
+    d = Dict(kwargs)
+
+    if haskey(d, :apiVersion)
+        v = d[:apiVersion]
+        k = Kuber.api_group_type(v)
+        apictx = k(ctx.client)
+        delete!(d, :apiVersion)
```

Comment:
Here too, better to take function arg.
And since this piece of code is common between `get` and `delete!` and also `put!` (already existing), better to extract this out into a function.

Reply:
I have updated for `update!` and `list` as well. I don't have any ways to test `update!`.

Reply:
Maybe test by using `update!` to reset the parallelism/replicas of a job/replicaset?